core/src/com/unciv/ui/utils/ZoomableScrollPane.kt

```diff
@@ -71,13 +71,35 @@ open class ZoomableScrollPane(
     }
 
     override fun scrollX(pixelsX: Float) {
-        super.scrollX(pixelsX)
+        val exploredRegion = UncivGame.Current.worldScreen?.viewingCiv?.exploredRegion
+        var result = pixelsX
+        if(exploredRegion != null && !exploredRegion.isEmpty()) {
+            val centerX = this.maxX * 0.5f + exploredRegion["tileRadius"]!!.x
+            var min = centerX+exploredRegion["xWorldVector"]!!.x
+            val max = centerX+exploredRegion["xWorldVector"]!!.y
+
+            if (result < min) result = min
+            else if (result > max) result = max
+        }
+
+        super.scrollX(result)
         updateCulling()
         onViewportChanged()
     }
 
     override fun scrollY(pixelsY: Float) {
-        super.scrollY(pixelsY)
+        val exploredRegion = UncivGame.Current.worldScreen?.viewingCiv?.exploredRegion
+        var result = pixelsY
+        if(exploredRegion != null && !exploredRegion.isEmpty()) {
+            val centerY = this.maxY * 0.5f
+            var min = centerY-exploredRegion["yWorldVector"]!!.x
+            val max = centerY-exploredRegion["yWorldVector"]!!.y
+
+            if (result < min) result = min
+            else if (result > max) result = max
+        }
+
+        super.scrollY(result)
         updateCulling()
         onViewportChanged()
     }
```

Comment:
I think this doesn't belong to `ZoomableScrollPane`, which is an abstract scroll pane object which in principle can contain everything, not only tile map. 

Moreover: CityScreen and MapEditorScreen which contain such scroll pane do not need camera limiting functionality: city screen has map size of like 8 tiles or so and map editor screen is not supposed to restrict vision of a map creator.

I'd propose to move this to `WorldMapHolder`, which is subclass of `ZoomableScrollPane` and already contains a reference to the current `WorldScreen`.